src/asm/charmap.cpp

```diff
@@ -42,12 +42,23 @@ bool charmap_ForEach(
     void (*mapFunc)(std::string const &),
     void (*charFunc)(std::string const &, std::vector<int32_t>)
 ) {
-	for (Charmap &charmap : charmapList) {
-		mapFunc(charmap.name);
-		for (size_t i = 0; i < charmap.nodes.size(); ++i) {
-			if (CharmapNode const &node = charmap.nodes[i]; node.isTerminal())
-				charFunc(charmap.mappings[i], node.value);
+	for (Charmap const &charmap : charmapList) {
+		// Traverse the trie depth-first to derive the character mappings in definition order
+		std::map<size_t, std::string> mappings;
+		for (std::stack<std::pair<size_t, std::string>> prefixes({{0, ""}}); !prefixes.empty();) {
+			auto [nodeIdx, mapping] = prefixes.top();
```

Comment:
Does this make a copy, or does this unpack two references?

Reply:
Well, we use `auto [...] = ...` everywhere else.

Reply:
RGBGFX, for example, has a number of `auto [k, v] = ...`.

Reply:
Right, but I'm concerned because we drop the element being unpacked right after.

Reply:
I can't measure a performance difference either way.

Reply:
I'm concerned about a potential UAF, not performance. Sorry if that wasn't clear.

Reply:
Oh! Then there isn't one, but to be really explicit about that (*and* avoid a copy), I added a `std::move` here.

Reply:
This does not solve the problem, because it's still a reference to the object that gets `pop`ped off, even if its a rvalue ref thanks to `std::move`.

Reply:
Either we disagree here or I'm misunderstanding you/C++, so please just commit whatever fix you have in mind.

Reply:
Okay, I found what I was looking for: `auto [nodeIdx, mapping]` copies the elements when unpacking, whereas `auto &[nodeIdx, mapping]` would produce references instead (which would become dangling after the `prefixes.pop()`.

So the current code is fine, at the cost of making a copy of the string (I *think* `std::move` makes it be a move-constructor copy? Idrk).

Reply:
It should, yes; and we saw performance difference was negligible anyway.